go/processor/floria/precompiled_test.go

```diff
@@ -80,3 +81,347 @@ func TestPrecompiled_AddressesAreHandledCorrectly(t *testing.T) {
+	}
+
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			state := tosca.NewMockWorldState(ctrl)
+			if test.err == nil {
+				state.EXPECT().GetNonce(tosca.Address{}).Return(uint64(5))
+				state.EXPECT().SetNonce(tosca.Address{}, uint64(5+increment))
+			}
+
+			gas, err := stateIncNonce(state, test.sender, test.input, test.gas)
+			if err != test.err {
+				t.Errorf("unexpected error, want %v, got %v", test.err, err)
+			}
+			if gas != 0 {
+				t.Errorf("unexpected gas, want %v, got %v", 0, gas)
+			}
+		})
+	}
+
+}
+
+func TestPrecompiled_HandleStatePrecompiled(t *testing.T) {
+	codeSize := tosca.Gas(42)
+	increment := byte(5)
+	tests := map[string]struct {
+		recipient          tosca.Address
+		inputPrefix        []byte
+		input              []byte
+		isStatePrecompiled bool
+		mockSetUp          func(*tosca.MockWorldState)
+	}{
+		"nonPrecompiled": {
+			tosca.Address{0x20},
+			[]byte{0x01, 0x02, 0x03, 0x04},
+			make([]byte, 64),
+			false,
+			func(mock *tosca.MockWorldState) {},
+		},
+		"setBalance": {
+			StateContractAddress,
+			[]byte{0xe3, 0x4, 0x43, 0xbc},
+			make([]byte, 64),
+			true,
+			func(mock *tosca.MockWorldState) {
+				mock.EXPECT().SetBalance(tosca.Address{}, tosca.Value{})
+			},
+		},
+		"copyCode": {
+			StateContractAddress,
+			[]byte{0xd6, 0xa0, 0xc7, 0xaf},
+			append(append(make([]byte, 12), 0x01), make([]byte, 19+32)...),
+			true,
+			func(mock *tosca.MockWorldState) {
+				mock.EXPECT().GetCode(tosca.Address{}).Return(make([]byte, codeSize))
+				mock.EXPECT().SetCode(tosca.Address{0x01}, tosca.Code(make([]byte, codeSize)))
+			},
+		},
+		"swapCode": {
+			StateContractAddress,
+			[]byte{0x7, 0x69, 0xb, 0x2a},
+			append(append(make([]byte, 12), 0x01), make([]byte, 19+32)...),
+			true,
+			func(mock *tosca.MockWorldState) {
+				mock.EXPECT().GetCode(tosca.Address{0x00}).Return(make([]byte, codeSize))
+				mock.EXPECT().GetCode(tosca.Address{0x01}).Return(make([]byte, codeSize))
+				mock.EXPECT().SetCode(tosca.Address{0x00}, tosca.Code(make([]byte, codeSize)))
+				mock.EXPECT().SetCode(tosca.Address{0x01}, tosca.Code(make([]byte, codeSize)))
+			},
+		},
+		"setStorage": {
+			StateContractAddress,
+			[]byte{0x39, 0xe5, 0x3, 0xab},
+			make([]byte, 96),
+			true,
+			func(mock *tosca.MockWorldState) {
+				mock.EXPECT().SetStorage(tosca.Address{}, tosca.Key{}, tosca.Word{})
+			},
+		},
+		"incNonce": {
+			StateContractAddress,
+			[]byte{0x79, 0xbe, 0xad, 0x38},
+			append(append(make([]byte, 32), increment), make([]byte, 31)...),
+			true,
+			func(mock *tosca.MockWorldState) {
+				mock.EXPECT().GetNonce(tosca.Address{}).Return(uint64(5))
+				mock.EXPECT().SetNonce(tosca.Address{}, uint64(5+increment))
+			},
+		},
+	}
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			state := tosca.NewMockWorldState(ctrl)
+			test.mockSetUp(state)
+
+			sender := DriverAddress
```

Comment:
Is there a test checking that the precompiled contract is not executed if the sender is != DriverAddress?